### PR TITLE
Sizing Calculator: added unit tests and minor restructuring

### PR DESCRIPTION
**Pull request by shaharuk-yb**
Associated with JIRA tasks: 
- https://yugabyte.atlassian.net/browse/DB-11648
- https://yugabyte.atlassian.net/browse/DB-11696
Includes following changes:
- Added unit tests to the sizing calculator
- restructured some code in sizing.go to support isolated run of the unit tests
- added github actions workflow to run unit tests

Function Under Test | Test name | Test scenario
-- | -- | --
getSourceMetadata | TestGetSourceMetadata_SuccessReadingSourceMetadata | successfully able to connect and load the source metadata
getSourceMetadata | TestGetSourceMetadata_QueryErrorIfTableDoesNotExistOrColumnsUnavailable | if table_index_stat does not exist in the assessment.db or one of the required column does not exist in the table
getSourceMetadata | TestGetSourceMetadata_RowScanError | verify if the all columns datatypes are correct. Expecting failure in case of unsupported data type. Example: expected column value is int but in assessment.db, float value is provided. Hence, it will fail to read/typecast
getSourceMetadata | TestGetSourceMetadata_NoRows | verify if there are no rows in the source metadata
shardingBasedOnTableSizeAndCount | TestShardingBasedOnTableSizeAndCount_TableWithSizePlacedInColocated | validate if the source table of size in colocated limit is correctly placed in colocated table recommendation
shardingBasedOnTableSizeAndCount | TestShardingBasedOnTableSizeAndCount_WithIndexes_ColocateAll | validate if the source table and index of size in colocated limit is correctly placed in colocated table recommendation
shardingBasedOnTableSizeAndCount | TestShardingBasedOnTableSizeAndCount_ColocatedLimitExceededBySize | validate if the source table and index of size in colocated limit exceeds the size limit and respective tables need to be put in sharded
shardingBasedOnTableSizeAndCount | TestShardingBasedOnTableSizeAndCount_ColocatedLimitExceededByCount | validate if the source table and index of size in colocated limit exceeds limit of maximum supported num objects and respective tables need to be put in sharded
shardingBasedOnTableSizeAndCount | TestShardingBasedOnTableSizeAndCount_NoColocatedTables | validate if the tables of size more than max colocated size supported are put in the sharded tables
shardingBasedOnOperations | TestShardingBasedOnOperations_CanSupportOpsRequirement | validate that the tables should be removed from colocated and added to sharded as the ops requirement is high
shardingBasedOnOperations | TestShardingBasedOnOperations_CannotSupportOpsAndNeedsSharding | validate that the tables should be removed from colocated and added to sharded as the ops requirement is higher than supported
checkShardedTableLimit | TestCheckShardedTableLimit_WithinLimit | validate that the sharded table limits is not exceeded and failure reasoning should be empty
checkShardedTableLimit | TestCheckShardedTableLimit_LimitExceeded | validate that the sharded table limits is exceeded and failure reasoning is generated as expected
findNumNodesNeededBasedOnThroughputRequirement | TestFindNumNodesNeededBasedOnThroughputRequirement_CanSupportOps | validate that the throughput requirements are supported and no scaling is needed
findNumNodesNeededBasedOnThroughputRequirement | TestFindNumNodesNeededBasedOnThroughputRequirement_NeedMoreNodes | validate that the throughput requirements cannot be supported by existing nodes and scaling is needed
findNumNodesNeededBasedOnTabletsRequired | TestFindNumNodesNeededBasedOnTabletsRequired_CanSupportTablets | validate that the tablets requirements are supported and no scaling is needed
findNumNodesNeededBasedOnTabletsRequired | TestFindNumNodesNeededBasedOnTabletsRequired_NeedMoreNodes | validate that the tablets cannot be supported by existing nodes and scaling is needed
pickBestRecommendation | TestPickBestRecommendation_PickOneWithOptimalNodesAndCores | validate if the recommendation with optimal nodes and cores is picked up
pickBestRecommendation | TestPickBestRecommendation_PickOneWithOptimalNodesAndCoresWhenSomeHasFailures | validate if the recommendation with optimal nodes is picked up when some of the recommendation has failure reasoning
pickBestRecommendation | TestPickBestRecommendation_PickLastMaxCoreRecommendationWhenNoneCanSupport | validate if the recommendation with optimal nodes and cores is picked up for showing failure reasoning
calculateTimeTakenAndParallelJobsForImport | TestCalculateTimeTakenAndParallelJobsForImport_ValidateFormulaToCalculateImportTime | validate the formula to calculate the import time
getReasoning | TestGetReasoning_NoObjects | validate reasoning when there are no objects(empty colocated and sharded objects)
getReasoning | TestGetReasoning_OnlyColocatedObjects | validate reasoning when there are only colocated objects
getReasoning | TestGetReasoning_OnlyShardedObjects | validate reasoning when there are only sharded objects
getReasoning | TestGetReasoning_ColocatedAndShardedObjects | validate reasoning when there are colocated and sharded objects
getReasoning | TestGetReasoning_Indexes | validate reasoning when there are colocated and sharded objects with indexes

### CODE REVIEW

**Review by makalaaneesh — APPROVED** [review]
Thanks for this @shaharuk-yb , this is very thorough!! 
Did not go through all the logic in detail, as you would be the better judge, but the overall structure looks good. 

Just had one comment, pls look into that before merging

File: yb-voyager/src/migassessment/sizing_test.go [hunk 1]
```diff
@@ -0,0 +1,907 @@
...
+	// Define test data
+	sourceIndexMetadata := []SourceDBMetadata{
+		{ObjectName: "table1", Size: 10.0, ReadsPerSec: 1000000, WritesPerSec: 50000000},
+	}
+	recommendation := map[int]IntermediateRecommendation{
+		2: {
+			ColocatedTables: []SourceDBMetadata{
+				{ObjectName: "table1", Size: 10.0, ReadsPerSec: 1000000, WritesPerSec: 50000000},
+			},
+			ShardedTables: []SourceDBMetadata{},
```
[the remark this answers is not in the document]
Comment (shaharuk-yb) [reply yb-voyager/src/migassessment/sizing_test.go:443]: It is as expected @makalaaneesh , the recommendations start of that way(table being colocated), and after the function under test runs, if is suppose go move from colocated to sharded and that's what we're checking in updatedRecomendation.

[the remark this answers is not in the document]
Comment (makalaaneesh) [reply yb-voyager/src/migassessment/sizing_test.go:443]: got it! I thought this was the expected recommendations, my bad.

File: yb-voyager/src/migassessment/sizing_test.go [hunk 2]
```diff
@@ -0,0 +1,907 @@
...
+	"errors"
+	"fmt"
+	"github.com/DATA-DOG/go-sqlmock"
+	"github.com/stretchr/testify/assert"
+	"math"
+	"testing"
+)
+
+var SourceDbSelectQuery = "SELECT schema_name, object_name, row_count, reads_per_second, writes_per_second, " +
+	"is_index, parent_table_name, size_in_bytes FROM table_index_stats ORDER BY size_in_bytes ASC"
```
[the remark this answers is not in the document]
Comment (shaharuk-yb) [reply yb-voyager/src/migassessment/sizing_test.go]: done. updated it.

File: yb-voyager/src/migassessment/sizing_test.go [hunk 3]
```diff
@@ -0,0 +1,907 @@
...
+	"database/sql"
+	"errors"
+	"fmt"
+	"github.com/DATA-DOG/go-sqlmock"
+	"github.com/stretchr/testify/assert"
+	"math"
+	"testing"
+)
+
+var SourceDbSelectQuery = "SELECT schema_name, object_name, row_count, reads_per_second, writes_per_second, " +
```
[the remark this answers is not in the document]
Comment (shaharuk-yb) [reply yb-voyager/src/migassessment/sizing_test.go]: updated :
- to ```AssessmentDbSelectQuery``` from previous ```SourceDbSelectQuery```
- to ```AssessmentDBColumns``` from previous ```SourceDBColumns```

**Pull request merged**